go/vt/vtctl/workflow/workflow_state_test.go

```diff
@@ -0,0 +1,167 @@
+// routing rules. We setup two workflows with the same table in both source and target keyspaces.
+func TestWorkflowStateMoveTables(t *testing.T) {
+	ctx := context.Background()
+	te := setupMoveTables(t, ctx)
+	require.NotNil(t, te)
+	type testCase struct {
+		name                   string
+		wf1SwitchedTabletTypes []topodatapb.TabletType
+		wf1ExpectedState       string
+		// Simulate a second workflow to validate that the logic used to determine the state of the first workflow
+		// from the routing rules is not affected by the presence of other workflows in different states.
+		wf2SwitchedTabletTypes []topodatapb.TabletType
+	}
+	testCases := []testCase{
+		{
+			name:                   "switch reads",
+			wf1SwitchedTabletTypes: []topodatapb.TabletType{topodatapb.TabletType_REPLICA, topodatapb.TabletType_RDONLY},
+			wf1ExpectedState:       "All Reads Switched. Writes Not Switched",
+			wf2SwitchedTabletTypes: []topodatapb.TabletType{topodatapb.TabletType_PRIMARY},
+		},
+		{
+			name:                   "switch writes",
+			wf1SwitchedTabletTypes: []topodatapb.TabletType{topodatapb.TabletType_PRIMARY},
+			wf1ExpectedState:       "Reads Not Switched. Writes Switched",
+			wf2SwitchedTabletTypes: []topodatapb.TabletType{topodatapb.TabletType_REPLICA, topodatapb.TabletType_RDONLY},
+		},
+		{
+			name:                   "switch reads and writes",
+			wf1SwitchedTabletTypes: []topodatapb.TabletType{topodatapb.TabletType_PRIMARY, topodatapb.TabletType_REPLICA, topodatapb.TabletType_RDONLY},
+			wf1ExpectedState:       "All Reads Switched. Writes Switched",
+		},
+		{
+			name:                   "switch rdonly only",
+			wf1SwitchedTabletTypes: []topodatapb.TabletType{topodatapb.TabletType_RDONLY},
+			wf1ExpectedState:       "Reads partially switched. Replica not switched. All Rdonly Reads Switched. Writes Not Switched",
+			wf2SwitchedTabletTypes: []topodatapb.TabletType{topodatapb.TabletType_PRIMARY},
+		},
+		{
+			name:                   "switch replica only",
+			wf1SwitchedTabletTypes: []topodatapb.TabletType{topodatapb.TabletType_REPLICA},
+			wf1ExpectedState:       "Reads partially switched. All Replica Reads Switched. Rdonly not switched. Writes Not Switched",
+			wf2SwitchedTabletTypes: []topodatapb.TabletType{topodatapb.TabletType_PRIMARY, topodatapb.TabletType_REPLICA, topodatapb.TabletType_RDONLY},
+		},
+	}
+	tables := []string{"t1"}
+
+	getStateString := func(targetKeyspace, wfName string) string {
+		tsw, state, err := te.ws.getWorkflowState(ctx, targetKeyspace, wfName)
+		require.NoError(t, err)
+		require.NotNil(t, tsw)
+		require.NotNil(t, state)
+		return state.String()
+	}
+	require.Equal(t, "Reads Not Switched. Writes Not Switched", getStateString("target", "wf1"))
+
+	resetRoutingRules := func() {
+		te.updateTableRoutingRules(t, ctx, nil, tables,
+			"source", te.targetKeyspace.KeyspaceName, "source")
+		te.updateTableRoutingRules(t, ctx, nil, tables,
+			"source2", "target2", "source2")
+	}
+	resetRoutingRules()
```

Comment:
We don't want to do this within t.Run? If we move this to the beginning of that function then we can remove the reset call at the end of it.